### ISSUE TITLE
Pause animations when rendering tests

### ISSUE THREAD

**Issue by Betree** [lost-pixel/lost-pixel#268]
### Mode

Page screenshots

### Feature description

Hi!

We're currently experimenting with Lost Pixel for https://github.com/opencollective/opencollective-frontend.

While the initial setup works mostly fine out of the box, we currently receive some false positives on animated elements - mostly loading spinners. Because the screenshots don't happen exactly at the same time, the spinner is in a different position when rendered.

It seems that Chromatic [pauses animation](https://www.chromatic.com/docs/animations) to address this.

## Examples

Styleguide: styleguide.opencollective.com

<details>
<summary>❗ storybook/design-system-loading-grid--default (click to expand)</summary>

| Before | After |
| --- | --- |
| ![](https://lp-00002-d34ac.fra1.digitaloceanspaces.com/cli4e0w6q2kjzmo0el8bu56i1/5484d427fa5a91235721c90d08ccf65de0e97b780807121126babbef1db696b5.png) | ![](https://lp-00001-a8c7a.fra1.digitaloceanspaces.com/cli4e0w6q2kjzmo0el8bu56i1/ac3c6a44468c3bf9ebe4fe49c990f1c15b1ae46d0fff75c4cff320ee38431cb0.png) |
</details>

<details>
<summary>❗ storybook/o-auth-application-approve-screen--auto-approve-redirecting (click to expand)</summary>

| Before | After |
| --- | --- |
| ![](https://lp-00001-a8c7a.fra1.digitaloceanspaces.com/cli4e0w6q2kjzmo0el8bu56i1/0f31e0bb5bdacf804137230dc086d715b72ec545d838cd1b00b67b8f79f12888.png) | ![](https://lp-00001-a8c7a.fra1.digitaloceanspaces.com/cli4e0w6q2kjzmo0el8bu56i1/9e27412c7d596ed3c57fef5b44b11f23e05766c4eda5c003c73ce228222950d6.png) |
</details>

Thank you!

**Comment by d-ivashchuk** [comment]
hey @Betree! thanks for using lost pixel! you could try to set it up in the correct way yourself - we didn't decide for the defaults but we might revise it. In beforeScreenshot you can prepare your browser in any possible way that you need.

```js
module.exports = {
  pageShots: {
    pagesJsonUrl: 'http://172.17.0.1:9000/lost-pixel.json',
    baseUrl: 'http://172.17.0.1:9000',
    mask: [{ selector: 'span.animated-image' }],
  },
  lostPixelProjectId: 'xxx',
  apiKey: process.env.LOST_PIXEL_API_KEY,
  compareConcurrency: 10,
  shotConcurrency: 10,
  threshold: 10,
  beforeScreenshot: async (page) => {
    await page.addStyleTag({
      content: `iframe {
          visibility: hidden;
        }

        /* do not show underline animation */
        #toc-holder  a {
          background-size: 0 !important;
          background-image: none !important;
        }
        /* skip image display within section */
        section img {
          visibility: hidden;
        }

        /* hide cookie banner */
        #onetrust-consent-sdk {
          display: none;
        }`,
    })
  },
}
```

**Comment by Betree** [comment]
@d-ivashchuk Good stuff, thank you! I initially took a similar approach, but injecting CSS based on env variable: [opencollective/opencollective-frontend@`691f1f3` (#8940)](https://github.com/opencollective/opencollective-frontend/pull/8940/commits/691f1f37ebf374dd922f5051a098df96614b264c)
I'll switch to the suggested approach as it seems cleaner.

Could be interesting to add a page about animations on https://docs.lost-pixel.com

**Comment by chriskalmar** [comment]
I'm surprised that this is not working properly. We have animations in a lot of projects and so far it wasn't an issue.

We do disable animations when taking screenshots:
https://github.com/lost-pixel/lost-pixel/blob/577ef49b36983a4252f7825fa7e4e3792b2838d2/src/shots/shots.ts#L119

This will only stop CSS animations, but not JS animations.
As it is a spinner I assume it's CSS, right?
@Betree

**Comment by Betree** [comment]
It turns out the problematic component was using SVG animations (which are deprecated). As part of the [config change](https://github.com/opencollective/opencollective-frontend/pull/8940), I also pushed a conversion of this component to CSS animations which is what actually resolved the issue.

Now [removing the custom config](https://github.com/opencollective/opencollective-frontend/pull/8948) for animations is not a breaking change, so you're absolutely correct @chriskalmar: CSS animations are supported out-of-the-box.

Thank you both very much for the support!

_TLDR ; if you're using SVG animations, either migrate (they're deprecated) to CSS animations or use a custom config to hide the whole thing._
